src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/MauiAddProjectPropertiesStep.cs

```diff
@@ -145,16 +148,33 @@ protected override Task<UpgradeStepInitializeResult> InitializeImplAsync(IUpgrad
 
             var project = context.CurrentProject.Required();
             var file = project.GetFile();
+            var components = await project.GetComponentsAsync(token).ConfigureAwait(false);
+            bool propertiesValueUpdated = true;
 
-            // var properties =  project.GetProjectPropertyElements();
-            if (string.IsNullOrEmpty(file.GetPropertyValue("UseMaui")))
+            // check project properties updated based on project types
+            if (components.HasFlag(ProjectComponents.Maui))
             {
-                Logger.LogInformation(".NET MAUI Project Properties need to be added.");
-                return Task.FromResult(new UpgradeStepInitializeResult(UpgradeStepStatus.Incomplete, ".NET MAUI Project Properties need to be added", BuildBreakRisk.High));
+                if (string.IsNullOrEmpty(file.GetPropertyValue("ImplicitUsings")))
+                {
+                    propertiesValueUpdated = false;
+                }
             }
             else
             {
-                return Task.FromResult(new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, ".NET MAUI Project Properties already added.", BuildBreakRisk.None));
+                if (string.IsNullOrEmpty(file.GetPropertyValue("UseMaui")))
+                {
+                    propertiesValueUpdated = false;
+                }
+            }
+
+            if (propertiesValueUpdated)
+            {
+                return Task.FromResult(new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, ".NET MAUI Project Properties already added.", BuildBreakRisk.None)).Result;
```

Comment:
I don't think you need/want the `.Result` on this
```suggestion
                return Task.FromResult(new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, ".NET MAUI Project Properties already added.", BuildBreakRisk.None));
```